### PR TITLE
Add a mapping of ArduPilot custom flight modes

### PR DESCRIPTION
**Pull request by WickedShell**
This maps out all currently supported flight modes that are reported via the `custom_mode` field of heartbeat.

### CODE REVIEW

**Comment by lvale** [comment]
ROVER_FLIGHT_MODE ROVER_FLIGHT_MODE_INITIALISING

I don't see the need for the repetition of the <VEHICLE>_FLIGHT_MODE part

**Comment by OXINARF** [comment]
@lvale The macro names don't have the enum name so if you remove the FLIGHT_MODE part, it will be just ROVER_INITIALISING which doesn't say it is a flight mode.

**Comment by WickedShell** [comment]
I can see the arguments either way. I suspect that `_FLIGHT` could be removed from the middle without losing any clarity, (and would actually make more sense for rover, sub, tracker without hurting copter or plane).

**Comment by WickedShell** [comment]
Renamed to <vehicle>_MODE_<mode>. If someone can just review that this matches the content of all the vehicles defines.h this should be good to go.

File: message_definitions/v1.0/ardupilotmega.xml [hunk 1]
```diff
@@ -918,6 +918,84 @@
...
+      <entry value="5"  name="PLANE_MODE_FLY_BY_WIRE_A"/>
+      <entry value="6"  name="PLANE_MODE_FLY_BY_WIRE_B"/>
+      <entry value="7"  name="PLANE_MODE_CRUISE"/>
+      <entry value="8"  name="PLANE_MODE_AUTOTUNE"/>
+      <entry value="10" name="PLANE_MODE_AUTO"/>
+      <entry value="11" name="PLANE_MODE_RTL"/>
+      <entry value="12" name="PLANE_MODE_LOITER"/>
+      <entry value="14" name="PLANE_MODE_AVOID_ADSB"/>
+      <entry value="15" name="PLANE_MODE_GUIDED"/>
+      <entry value="16" name="PLANE_MODE_INITIALISING"/>
```
Comment (peterbarker) [inline message_definitions/v1.0/ardupilotmega.xml]: Shold be `INITIALIZING` for consistency.

File: message_definitions/v1.0/ardupilotmega.xml [hunk 2]
```diff
@@ -918,6 +918,84 @@
...
       </entry>
     </enum>
+    <enum name="PLANE_MODE">
+      <description>A mapping of plane flight modes for custom_mode field of heartbeat</description>
+      <entry value="0"  name="PLANE_MODE_MANUAL"/>
+      <entry value="1"  name="PLANE_MODE_CIRCLE"/>
+      <entry value="2"  name="PLANE_MODE_STABILIZE"/>
+      <entry value="3"  name="PLANE_MODE_TRAINING"/>
+      <entry value="4"  name="PLANE_MODE_ACRO"/>
+      <entry value="5"  name="PLANE_MODE_FLY_BY_WIRE_A"/>
```
Comment (peterbarker) [inline message_definitions/v1.0/ardupilotmega.xml]: Maybe a little wordy.

File: message_definitions/v1.0/ardupilotmega.xml (hunk 1 above)
Comment (WickedShell) [reply message_definitions/v1.0/ardupilotmega.xml]: Fixed.

File: message_definitions/v1.0/ardupilotmega.xml (hunk 2 above)
Comment (WickedShell) [reply message_definitions/v1.0/ardupilotmega.xml]: I'd prefer the clarity/lack of ambiguity of spelling it out here.

**Comment by OXINARF** [comment]
Has anyone confirmed this matches the ArduPilot values?

**Comment by peterbarker** [comment]
On Thu, 25 Jan 2018, Francisco Ferreira wrote:

> Has anyone confirmed this matches the ArduPilot values?

I went through them, and didn't see anything different.

**Comment by OXINARF** [comment]
Thanks @peterbarker! Merged!

**Pull request merged**